sbor/src/type_id.rs

```diff
@@ -147,6 +148,13 @@ impl<T: TypeId> TypeId for Box<T> {
     }
 }
 
+impl<T: TypeId> TypeId for Rc<T> {
```

Comment:
Yes, encoding and decoding of Rc and Arc may introduce multiple copies of the same data.  This is okay for our system, since only vaults are eventually persisted (and only one copy exists).

This is also consistent with other serialization framework, like Serde. 

A better approach would be to make it a feature, so external users of SBOR have the option to turn it off.